From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: SeongJae Park <sj@kernel.org>, damon@lists.linux.dev, kernel-team@meta.com, linux-kernel@vger.kernel.org, linux-mm@kvack.org
Message-Id: <20250521042755.39653-2-sj@kernel.org>
In-Reply-To: <20250521042755.39653-1-sj@kernel.org>
Date: Tue, 20 May 2025 21:27:54 -0700
Subject: [PATCH 1/2] mm/damon/Kconfig: set DAMON_{VADDR,PADDR,SYSFS} default to DAMON

DAMON_{VADDR,PADDR,SYSFS} are de-facto essential parts of DAMON for
normal usages.  Because those need to be enabled one by one, however,
and there are other test-purpose or non-essential configurations, it is
easy to be confused and make mistakes at setup.  Make the essential
configurations default to CONFIG_DAMON, so that those can be enabled by
default with a single change.

Signed-off-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20250521042755.39653-2-sj@kernel.org
Cc: damon@lists.linux.dev
Cc: kernel-team@meta.com
Cc: linux-kernel@vger.kernel.org
Cc: linux-mm@kvack.org
---
 mm/damon/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index c213cf8b5638..c93d0c56b963 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -28,6 +28,7 @@ config DAMON_VADDR
 	bool "Data access monitoring operations for virtual address spaces"
 	depends on DAMON && MMU
 	select PAGE_IDLE_FLAG
+	default DAMON
 	help
 	  This builds the default data access monitoring operations for DAMON
 	  that work for virtual address spaces.
@@ -36,6 +37,7 @@ config DAMON_PADDR
 	bool "Data access monitoring operations for the physical address space"
 	depends on DAMON && MMU
 	select PAGE_IDLE_FLAG
+	default DAMON
 	help
 	  This builds the default data access monitoring operations for DAMON
 	  that works for the physical address space.
@@ -55,6 +57,7 @@ config DAMON_VADDR_KUNIT_TEST
 config DAMON_SYSFS
 	bool "DAMON sysfs interface"
 	depends on DAMON && SYSFS
+	default DAMON
 	help
 	  This builds the sysfs interface for DAMON.  The user space can use
 	  the interface for arbitrary data access monitoring.
-- 
2.39.5